### PR TITLE
Views ancestors for hdr

### PR DESCRIPTION
**Pull request by servantftransperfect**
Each view has now a list of ancestor views. It is an array of IndexT which are the views (may not exist anymore in this sfmData) from which the current view was created. For example a HDR view may have all the LDR views as its ancestors.

### CODE REVIEW

File: src/aliceVision/sfmDataIO/AlembicImporter.cpp [hunk 1]
```diff
@@ -329,7 +329,7 @@ bool readPointCloud(const Version & abcVersion, IObject iObj, M44d mat, sfmData:
 
     }
     else {
-        ALICEVISION_LOG_WARNING("Alembic LOAD: NO OBSERVATIONS_WITH_FEATURES: "
+        ALICEVISION_LOG_WARNING("Alembic hb: NO OBSERVATIONS_WITH_FEATURES: "
```
Comment (fabiencastan) [inline src/aliceVision/sfmDataIO/AlembicImporter.cpp:332]: ?

**Review by fabiencastan — CHANGES_REQUESTED** [review]
Move localBA commit in a separate PR.

File: src/aliceVision/sfmDataIO/jsonIO.cpp [hunk 2]
```diff
@@ -58,6 +58,20 @@ void saveView(const std::string& name, const sfmData::View& view, bpt::ptree& pa
     viewTree.add_child("metadata", metadataTree);
   }
 
+  // metadata
```
Comment (fabiencastan) [inline src/aliceVision/sfmDataIO/jsonIO.cpp:61]: Maybe add:
`if(!view.getAncestors().empty())`

File: src/aliceVision/sfmData/View.hpp [hunk 3]
```diff
@@ -702,6 +702,24 @@ class View
...
+   * @brief Add a new ancestor to this view
+   * @param[in] viewId the view ancestor id
+   */
+  void addAncestor(IndexT viewId)
+  {
+    _ancestors.push_back(viewId);
+  }
+
+  /**
+  * @Brief get all ancestors for this view
```
Comment (fabiencastan) [inline src/aliceVision/sfmData/View.hpp:715]: Explain ancestor notion in documentation, like
```
@brief Get the list of viewID referencing the source views called "Ancestors"
If an image is generated from multiple input images, "Ancestors" allows to keep track of the viewIDs of the original inputs views.
For instance, the generated view can come from the fusion of multiple LDR images into one HDR image, the fusion from multi-focus stacking to get a fully focused image, fusion of images with multiple lighting to get a more diffuse lighting, etc.
@return list of viewID of the ancestors
```

Comment (servantftechnicolor) [reply src/aliceVision/sfmData/View.hpp:715]: applied

File: src/aliceVision/sfmDataIO/jsonIO.cpp (hunk 2 above)
Comment (servantftechnicolor) [reply src/aliceVision/sfmDataIO/jsonIO.cpp:61]: done

File: src/aliceVision/sfmDataIO/AlembicImporter.cpp (hunk 1 above)
Comment (fabiencastan) [reply src/aliceVision/sfmDataIO/AlembicImporter.cpp:332]: seems to be still there in the github global diff

**Comment by fabiencastan** [comment]
It would be good to increment the minor version for ABC, SFM formats, if we ever need to identify this version.

File: src/aliceVision/sfmDataIO/AlembicImporter.cpp (hunk 1 above)
Comment (servantftechnicolor) [reply src/aliceVision/sfmDataIO/AlembicImporter.cpp:332]: ...

**Pull request merged**